src/other/utils.jl

```diff
@@ -19,12 +15,12 @@ function identifier(s::AbstractString)
 end
 
 function makeidentifier(s::AbstractString)
-    i = start(s)
-    done(s, i) && return "x"
+    i = firstindex(s)
```

Comment:
This is not completely correct AFAIK, as the iteration state isn't required to be the index. We should use `iterate(s)` instead, and do `i === nothing && return "x"`. Then below you also need to check `i === nothing` rather than `i <= ncodeunits(s)`.

Reply:
These are the changes suggested by the deprecation messages.  I tried to alter the code according to your suggestions and was unable to get a version that passed tests.  It may be better to revert to the original version as I am not confident in my changes.  I don't have a clear idea of the logic in this code.

Reply:
Ah, you're right that the deprecation suggests this. But I'm not sure whether it's supposed to be a general solution, or whether it's just because there's no better deprecation for `start(::AbstractString)` as a temporary measure. Anyway, we really shouldn't need to use `firstindex` here.

Doesn't something like this work?
```julia
i = iterate(s)
i === nothing && return "x"
c, i = iterate(s, i)
...
while i !== nothing
    c, i = iterate(s, i)

Reply:
On thinking about it some more, it isn't `i` that gets checked for `=== nothing`.  I believe it is more like
```julia
(iresult = iterate(s)) === nothing && return "x"
(c, i) = iresult
...
while (iresult = iterate(s, i)) !== nothing
    (c, i) = iresult
```
I'll try that out